.github/workflows/ci.yml

```diff
@@ -127,3 +128,22 @@ jobs:
         uses: actions-rs/cargo@v1
         with:
           command: build
+
+  miri:
+    name: Test with Miri
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
+
+      - name: Install Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: nightly
+          components: miri
+          override: true
+
+      - name: Test
+        run: MIRIFLAGS="-Zmiri-tag-raw-pointers -Zmiri-check-number-validity" cargo miri test
```

Comment:
I assume you'll know best: what's the likelihood of these changing and causing our CI to fail?

Reply:
These flags basically put Miri into "hard mode", adding some extra checks that are off-by-default but give extra confidence in the unsafe code.  I'm not sure what kind of `unsafe` code httparse uses or which kind of code you plan to use in the future, so I cannot really give failure probabilities for the UB checks.

When we change flags we usually first just deprecate them (showing a warning) because we know removing them would break CI. Even this happens rarely. The last actual breaking change wrt our flags was https://github.com/rust-lang/miri/pull/1769; back then I made an effort to send PRs to all projects I could find that used Miri on CI to fix them preemptively. I don't even remember the last breaking flag change before that...

Reply:
Sounds good to me, thanks!